Storage/src/Connection/Rest.php

```diff
@@ -431,7 +431,11 @@ private function resolveUploadOptions(array $args)
         }
 
         $args['metadata']['name'] = $args['name'];
+        if (isset($args['retention'])) {
+            $args['metadata']['retention'] = $args['retention'];
```

Comment:
Why do we move this into the metadata instead of keeping it in the request body? A comment would be helpful here...

Reply:
```suggestion
            // during object creation retention properties go into metadata
            // but not into request body
            $args['metadata']['retention'] = $args['retention'];
```